samza-kafka/src/test/java/org/apache/samza/system/kafka/TestKafkaSystemFactoryJava.java

```diff
@@ -45,15 +46,13 @@ public void testGetIntermediateStreamProperties() {
     // no properties for stream
     config.put("streams.test.samza.intermediate", "true");
     config.put("streams.test.compression.type", "lz4"); //some random config
-    properties = JavaConversions.mapAsJavaMap(
-        factory.getIntermediateStreamProperties(new MapConfig(config)));
-    assertTrue(properties.isEmpty());
```

Comment:
I think these lines should be able stay here, since they should be filtered out as well but kafkaStreamSpec?

Reply:
Good catch! I added them back.